approvaltests/namer/stack_frame_namer.py

```diff
@@ -40,7 +42,25 @@ def get_test_frame_index(caller: List[FrameInfo]) -> int:
         raise FrameNotFound(message)
 
     @staticmethod
-    def is_test_method(frame: FrameInfo) -> bool:
+    def is_pytest_test(method_name: str) -> bool:
+        patterns = PytestConfig.test_naming_patterns
+        
+        # taken from pytest/python.py (class PyCollector)
+        for pattern in patterns:
+            if method_name.startswith(pattern):
+                return True
+                # Check that name looks like a glob-string before calling fnmatch
+                # because this is called for every name in each collected module,
+                # and fnmatch is somewhat expensive to call.
+            elif ("*" in pattern or "?" in pattern or "[" in pattern) and fnmatch.fnmatch(
```

Comment:
Is this the same code that pytest uses internally? Is there any way to call the same code in pytest as a library?

Reply:
Indeed, good eye.
See discussion [here](https://github.com/approvals/ApprovalTests.Python/pull/157#issuecomment-1884369703).